src/CSV/CsvManager.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace App\CSV;
+
+use App\DTO\DTOImportData;
+use App\Interfaces\CSV\CsvManagerInterface;
+use App\Interfaces\CSV\DataImportManagerInterface;
+use App\Interfaces\CSV\ImportResultInterface;
+use App\Repository\DonationRepository;
+use App\Repository\PersonRepository;
+use App\Repository\ProjectRepository;
+use App\Repository\RewardRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\Serializer\SerializerInterface;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+final class CsvManager implements CsvManagerInterface
+{
+    public function __construct(
+        private EntityManagerInterface $entityManager,
+        private SerializerInterface $serializer,
+        private PersonRepository $personRepository,
+        private DonationRepository $donationRepository,
+        private ProjectRepository $projectRepository,
+        private RewardRepository $rewardRepository,
+        private DataImportManagerInterface $dataImportManagerInterface,
+        private ValidatorInterface $validator
+    ) {
+    }
+
+    public function import(\SplFileInfo $filePath): ImportResultInterface
+    {
+        $errorsString = $this->importData($this->getDataFromFile($filePath->getPathname()), $this->validator);
+
+        $persons = $this->personRepository->findAll();
+        $donations = $this->donationRepository->findAll();
+        $rewards = $this->rewardRepository->findAll();
+        $projects = $this->projectRepository->findAll();
+
+        return new ImportResult($persons, $donations, $rewards, $projects, $errorsString);
+    }
+
+    private function importData(array $data, ValidatorInterface $validator): ?string
+    {
+        $errorsString = null;
+
+        foreach ($data as $row) {
+
+            $DTOImportData =
+                new DTOImportData(
+                    $row['first_name'],
+                    $row['last_name'],
+                    $row['project_name'],
+                    $row['reward'],
+                    $row['reward_quantity'],
+                    $row['amount']
+                );
+
+            $errors = $validator->validate($DTOImportData);
+
+            if (count($errors) > 0) {
+                $errorsString = (string) $errors;
+            }
```

Comment:
If there are some validation error, why you do import ? for me if there are some error, the line can't be import.